dashboard/test/controllers/api_controller_test.rb

```diff
@@ -59,79 +66,73 @@ def create_script_with_lockable_stage
     [script, level, stage]
   end
 
-  def make_progress_in_section(script)
-    text_response_script_levels = script.script_levels.includes(:level).where('levels.type' => TextMatch)
-
-    level = text_response_script_levels.first.level
-    UserLevel.create(level_id: level.id, user_id: @student_1.id, script_id: script.id)
+  def make_text_progress_in_script(script, student)
+    level = script.script_levels.map(&:oldest_active_level).find {|l| l.is_a? TextMatch}
+    level_source = create :level_source
+    create :user_level, level: level, user: student, script: script, level_source: level_source
+    # UserLevel.create!(level_id: level.id, user_id: student.id, script_id: script.id, level_source: level_source)
   end
 
   test "should get text_responses for section with default script" do
     get :section_text_responses, params: {section_id: @section.id}
     assert_response :success
 
-    assert_equal Script.twenty_hour_script, assigns(:script)
+    # we fall back to twenty_hour_script, which has no text_response levels
+    assert_equal '[]', @response.body
   end
 
   test "should get text_responses for section with section script" do
-    get :section_text_responses, params: {section_id: @flappy_section.id}
+    make_text_progress_in_script(@allthings_section.script, @student_allthings)
+
+    get :section_text_responses, params: {section_id: @allthings_section.id}
     assert_response :success
 
-    assert_equal Script.get_from_cache(Script::FLAPPY_NAME), assigns(:script)
+    response = JSON.parse(@response.body)
+
+    # make sure our response has stage from allthethingsscript
+    assert /\/s\/allthethings\// =~ response[0]['url']
```

Comment:
Oh sorry, I put the comment in the wrong place. It looks like just `make_text_progress_in_script`.